Minio/MinioClient.cs

```diff
@@ -529,6 +529,9 @@ public MinioClient WithCredentialsProvider(ClientProvider provider)
             responseResult = new ResponseResult(request, response);
             if (requestMessageBuilder.ResponseWriter != null)
                 requestMessageBuilder.ResponseWriter(responseResult.ContentStream);
+            if (requestMessageBuilder.FunctionResponseWriter != null)
+                requestMessageBuilder.ProcessFunctionResponseWriter(requestMessageBuilder.FunctionResponseWriter,
```

Comment:
why not simply
`await requestMessageBuilder.FunctionResponseWriter(responseResult.ContentStream)`
?

Reply:
@codemnk ,

Belated thanks for your review comments.
I'll create another PR for your suggested changes when we determine them to be good.

About your comment:
Response writer is not always populated, so we have to check it. Hence we cannot simply make a call to FunctionResponseWriter.

Reply:
@ebozduman,

I was not suggesting removing null check, I was talking about next line:
instead of using "task.Wait()" (sync call) you better use "await task" (async call)

Reply:
ok.
Thanks.
The new PR for your review comments is #669.